src/porepy/viz/exporter.py

```diff
@@ -131,56 +132,270 @@ def __init__(
+        # 2. Check whether the vtu file is compatible with the corresponding grids.
+        # 3. Transfer data from vtu to the mixed-dimensional grid.
+        for i, file_name in enumerate(file_names):
+
+            # Read all data from vtu
+            vtu_data = meshio.read(file_name)
+
+            # 1st step: Determine dimension of the grid associated to vtu file, and
+            # whether the grid corresponds to a subdomain or interface.
+
+            # Allow for automatic detection from the file name.
+            if kwargs.pop("automatic", True):
+                # Assuming grid_dim and is_mortar are among the stored vtu data
+                # one could utilize those, but this does not have to be the case.
+                # Thus, use naming convention of the Exporter for this.
+
+                # Remove ending '.vtu' from file_name, and decompose into pieces
+                # separated by '_'.
+                file_name_pieces = Path(file_name).stem.split("_")
+
+                # If the last two are numbers, the first one denotes the dimension.
+                assert file_name_pieces[-1].isnumeric()
+                dim_pos = -2 if file_name_pieces[-2].isnumeric() else -1
+                dim = int(file_name_pieces[dim_pos])
+
+                # If the key words before are 'mortar', or 'mortar' and 'constant',
+                # the vtu file corresponds to intefaces; otherwise to subdomains.
+                is_subdomain_data = not (
+                    file_name_pieces[dim_pos - 1] == "mortar"
+                    or file_name_pieces[dim_pos - 1] == "constant"
+                    and file_name_pieces[dim_pos - 2] == "mortar"
+                )
+
+            else:
+                # Read from keyword arguments
+
+                # Dimensionality of the grid
+                if "dims" not in kwargs:
+                    raise ValueError(
+                        """Provide dimensionality of the grids associated
+                        to the vtu files."""
+                    )
+                dims = kwargs.pop("dims")
+                assert isinstance(dims, list) or isinstance(dims, int)
+                dim = dims[i] if isinstance(dims, list) else dims
+
+                # Whether the grid is a subdomain or interface
+                if "are_subdomain_data" not in kwargs:
+                    is_subdomain_data = True
+                else:
+                    are_subdomain_data = kwargs.pop("are_subdomain_data")
+                    assert isinstance(are_subdomain_data, bool) or isinstance(
+                        are_subdomain_data, list
+                    )
+                    is_subdomain_data = (
+                        are_subdomain_data[i]
+                        if isinstance(are_subdomain_data, list)
+                        else are_subdomain_data
+                    )
+
+            # 2nd step: Make sure that the vtu file and the current grid are identical.
```

Comment:
Please specify what is meant by identical: Should the number of cells, faces etc. be the same, or also coordinates.

Reply:
I made it more concrete and detailed.